pallets/inflation/src/tests.rs

```diff
@@ -0,0 +1,473 @@
+
+        // Execute call, ensure it works
+        assert_ok!(Inflation::force_set_inflation_config(
+            RuntimeOrigin::root(),
+            new_config
+        ));
+        System::assert_last_event(
+            Event::InflationConfigurationForceChanged { config: new_config }.into(),
+        );
+
+        assert_eq!(ActiveInflationConfig::<Test>::get(), new_config);
+    })
+}
+
+#[test]
+fn force_set_inflation_config_fails() {
+    ExternalityBuilder::build().execute_with(|| {
+        let mut new_config = ActiveInflationConfig::<Test>::get();
+        new_config.recalculation_block = new_config.recalculation_block + 50;
+
+        // Make sure action is privileged
+        assert_noop!(
+            Inflation::force_set_inflation_config(RuntimeOrigin::signed(1), new_config),
+            BadOrigin
+        );
+    })
+}
+
+#[test]
+fn force_inflation_recalculation_work() {
+    ExternalityBuilder::build().execute_with(|| {
+        let old_config = ActiveInflationConfig::<Test>::get();
+
+        // Execute call, ensure it works
+        assert_ok!(Inflation::force_inflation_recalculation(
+            RuntimeOrigin::root(),
+        ));
+
+        let new_config = ActiveInflationConfig::<Test>::get();
+        assert!(
+            old_config != new_config,
+            "Config should change, otherwise test doesn't make sense."
+        );
+
+        System::assert_last_event(
+            Event::ForcedInflationRecalculation { config: new_config }.into(),
+        );
+    })
+}
+#[test]
+fn force_inflation_fails_due_to_unprivileged_origin() {
+    ExternalityBuilder::build().execute_with(|| {
+        // Make sure action is privileged
+        assert_noop!(
+            Inflation::force_inflation_recalculation(RuntimeOrigin::signed(1)),
+            BadOrigin
+        );
+    })
+}
+
+#[test]
+fn inflation_recalculation_occurs_when_exepcted() {
+    ExternalityBuilder::build().execute_with(|| {
+        let init_config = ActiveInflationConfig::<Test>::get();
+
+        // Make sure `on_finalize` calls before the expected change are storage noops
+        advance_to_block(init_config.recalculation_block - 3);
+        assert_storage_noop!(Inflation::on_finalize(init_config.recalculation_block - 3));
+        Inflation::on_initialize(
+            init_config.recalculation_block - 2
+        );
+        assert_storage_noop!(Inflation::on_finalize(init_config.recalculation_block - 2));
+        Inflation::on_initialize(
+            init_config.recalculation_block - 1
+        );
+
+        // One block before recalculation, on_finalize should calculate new inflation config
+        let init_config = ActiveInflationConfig::<Test>::get();
+        let init_total_issuance = Balances::total_issuance();
+
+        // Finally trigger inflation recalculation.
+        Inflation::on_finalize(init_config.recalculation_block - 1);
+
+        let new_config = ActiveInflationConfig::<Test>::get();
+        assert!(
```

Comment:
assert_ne!(new_config, init_config, "Recalculation must happen at this point.");